### PR TITLE
Use golden files for converter testings

### PR DESCRIPTION
**Pull request by mustard-mh**
## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 382f865</samp>

This pull request adds various golden and JSON files for testing different functions in the `typescript-common` module. The functions are related to converting between different types of objects, such as org member roles, configurations, and auth providers.

</details>

## Related Issue(s)

Fixes #

## How to test

- json file and golden file pairs should align
- `yarn test` should works
```
cd components/public-api/typescript-common
yarn build && yarn test

# update some golden files under `fixtures` folder
yarn test # should failed
```

| ✅ result|
|:-:|
|image|

#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/public-api/typescript-common/package.json
Comment (akosyakov) [inline components/public-api/typescript-common/package.json:20]: @mustard-mh it would be good to add README with how to do it and share on public api channel

**Review by akosyakov — APPROVED** [review]
to unblock, but please do comms

**Comment by filiptronicek** [comment]
@mustard-mh can you please add the generated files to https://github.com/gitpod-io/gitpod/blob/main/.gitattributes 🙏?

**Comment by filiptronicek** [comment]
@mustard-mh I think they will not be completely hidden, but just not expanded by default. This means that when reviewing PRs, other files will be easier to find

This is how it looks like for generated files for the public API.

image

File: components/public-api/typescript-common/package.json
Comment (mustard-mh) [reply components/public-api/typescript-common/package.json:20]: > one question will be how to iterate on a single test?

Do you mean to force update only one test? After exec this command, contributor and reviewer has responsibility to verify that golden files are expected. 

It's not just run it then that's ok, we want to have a constant output in converter

**Comment by mustard-mh** [comment]
@filiptronicek  I thought we want to them be displayed by default? 

(I feel it's easier to compare the results in github's PR diff view than in Editors, but not sure about others, i.e. you could try to review input json and golden files, then maybe you have the same feeling)

> contributor and reviewer has responsibility to verify that golden files are expected. [[src](https://github.com/gitpod-io/gitpod/pull/19168#discussion_r1410895225)]

**Comment by filiptronicek** [comment]
@mustard-mh if we will want to review changes to the files it indeed does not make too much sense. Your call 🙏

**Comment by filiptronicek** [comment]
btw many thanks for the README, love to see it!

**Pull request merged**
